src/scm/github.js

```diff
@@ -0,0 +1,433 @@
+      return response;
+    })
+    .then(getGithubRepos)
+    .then(updateRepo)
+    .then(updateBranch)
+    .then(() => {
+      $('#new-repo-name').val('');
+      $('#new-repo-desc').val('');
+      $('#new-repo-type').val('public');
+      showAlert(`Successfully create new repository ${repo}`);
+    })
+    .catch((err) => {
+      showAlert('Failed to create new repository.', LEVEL_ERROR);
+    });
+  }
+
+  createGist() {
+    const desc = $('#new-gist-name').val();
+    const isPublic = $('#new-gist-public').val() !== 'secret';
+    const payload = {
+      'description' : desc,
+      'public': isPublic,
+      'files': {
+        'init_by_gas_hub.html' : {
+          'content': 'init by gas-hub, just delete this file.'
+        }
+      }
+    };
+
+    new Promise((resolve, reject) => {
+      $.ajax({
+        url: `${this.baseUrl}/gists`,
+        headers: {
+          'Authorization': `token ${this.accessToken}`
+        },
+        method: 'POST',
+        crossDomain: true,
+        dataType: 'json',
+        contentType: 'application/json',
+        data: JSON.stringify(payload)
+      })
+      .then(resolve)
+      .fail(reject);
+    })
+    .then((response) => {
+      const gist = response.id;
+      context.branch = gist;
+      Object.assign(context.bindBranch, { [context.id] : gist });
+      chrome.storage.sync.set({ bindBranch: context.bindBranch });
+      return response;
+    })
+    .then(updateGist)
+    .then(() => {
+      $('#new-gist-name').val('');
+      $('#new-gist-public').val('public');
+      showAlert(`Successfully create new gist.`);
+    })
+    .catch((err) => {
+      showAlert('Failed to create new gist.', LEVEL_ERROR);
+    });
+  }
+
+  createBranch() {
+    const branch = $('#new-branch-name').val();
+    if (!branch || branch === '') return;
+    new Promise((resolve, reject) => {
+      $.getJSON(
+        `${this.baseUrl}/repos/${context.repo.fullName}/git/refs/heads/master`,
+        { access_token: this.accessToken }
+      )
+      .then(resolve)
+      .fail(reject)  
+    })
+    .then((response) => {
+      if (response.object) {
+        return response.object.sha;
+      }
+      else {
+        return $.getJSON(
+          `${this.baseUrl}/repos/${context.repo.fullName}/git/refs/heads`,
+          { access_token: this.accessToken }
+        )
+        .then((response) => {
+          return response[0].object.sha;
+        })
+      }
+    })
+    .then((sha) => {
+      const payload = {
+        ref: `refs/heads/${branch}`,
+        sha: sha
+      };
+      return $.ajax({
+        url: `${this.baseUrl}/repos/${context.repo.fullName}/git/refs`,
+        headers: {
+          'Authorization': `token ${this.accessToken}`
+        },
+        method: 'POST',
+        crossDomain: true,
+        dataType: 'json',
+        contentType: 'application/json',
+        data: JSON.stringify(payload)
+      });
+    })
+    .then((response) => {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/65105829498a48107e9a8b01cb705b310fffdc68/src/scm/github.js#L414"></span>‘response’ is defined but never used.</td>
    </tr>
  </tbody>
</table>